compiling with musl libc fails due to backtrace() being available in glibc only

***PCSX2 version:***
``latest git (67a6f31b8828194269aec7ac030e00948200ab20)``

***Description of the issue:***
``the wxStackWalker feature of wxWidgets relies on the gnu extension backtrace(), and as such is not available on systems with libc other than glibc. This makes compilation fail.``

***How to reproduce the issue:***
``try to compile on a linux distribution that utilizes musl libc (e.g Alpine Linux)``

from  <wx/stackwalk.h> :
```
#if wxUSE_STACKWALKER

class WXDLLIMPEXP_FWD_BASE wxStackFrame;
```
and I get
```
/home/pmos/pcsx2/pcsx2/gui/AppAssert.cpp:43:34: error: 'wxStackFrame' does not name a type; did you mean 'GtkFrame'?
  virtual void OnStackFrame(const wxStackFrame& frame)
```
so wxUSE_STACKWALKER is probably set to 0 because wxWidgets have figured out during compilation that backtrace() isn't a thing

```
/home/pmos/pcsx2/pcsx2/gui/AppAssert.cpp:43:34: error: 'wxStackFrame' does not name a type; did you mean 'GtkFrame'?
```
still.

because of:
```
#if wxUSE_STACKWALKER

class WXDLLIMPEXP_FWD_BASE wxStackFrame;
```
still

--- a/pcsx2/gui/AppAssert.cpp
+++ b/pcsx2/gui/AppAssert.cpp
@@ -19,6 +19,7 @@
 
 #include <wx/stackwalk.h>
 
+#if wxUSE_STACKWALKER
 class StackDump : public wxStackWalker
 {
 protected:
@@ -94,6 +95,7 @@ static wxString pxGetStackTrace( const FnChar_t* calledFrom )
 	dump.Walk( 3 );
 	return dump.GetStackTrace();
 }
+#endif
 
 #ifdef __WXDEBUG__
 
@@ -122,7 +124,11 @@ bool AppDoAssert( const DiagnosticOrigin& origin, const wxChar *msg )
 	static bool disableAsserts = false;
 	if( disableAsserts ) return false;
 
+#if wxUSE_STACKWALKER
 	wxString trace( pxGetStackTrace(origin.function) );
+#else
+	wxString trace( "Warning: Platform doesn't support wx stackwalker" );
+#endif
 	wxString dbgmsg( origin.ToString( msg ) );
 
 	wxMessageOutputDebug().Printf( L"%s", WX_STR(dbgmsg) );

